src/test/scala/com/campudus/tableaux/LinkTest.scala

```diff
@@ -209,6 +199,173 @@ class LinkTest extends TableauxTestBase {
     }
   }
 
+  private def setupTablesWithEmptyLinks(): Future[Number] = {
+    val linkColumn = Json.obj(
+      "columns" -> Json.arr(
+        Json.obj(
+          "name" -> "Test Link 1",
+          "kind" -> "link",
+          "fromColumn" -> 1,
+          "toTable" -> 2,
+          "toColumn" -> 1
+        )
+      )
+    )
+
+    def addRow(tableId: Long, values: JsonObject): Future[Number] = for {
+      res <- sendRequestWithJson("POST", values, s"/tables/$tableId/rows")
+      table1RowId1 <- Future.apply(res.getArray("rows").get[JsonObject](0).getNumber("id"))
+    } yield table1RowId1
+
+    def valuesRow(c: String) = Json.obj(
+      "columns" -> Json.arr(Json.obj("id" -> 1), Json.obj("id" -> 2)),
+      "rows" -> Json.arr(Json.obj("values" -> Json.arr(c, 2)))
+    )
+
+    for {
+    // setup two tables
```

Comment:
No, not strange indentation. Method definition in a method. You recommended this. ;-)